### DISCUSSION TITLE
Does zustand Internally maintains mutable state?

### DISCUSSION THREAD

**Discussion by cobra0411** [pmndrs/zustand#2719] [Q&A] [answered]
On the docs, It mentioned that zustand states are immutable. Can anyone share docs of internal working of states how exactly state are updated?

**Comment by dai-shi** [comment, marked as answer]
state objects are immutable, but the state itself is mutable internally. You can only update it with `setState`.
https://github.com/pmndrs/zustand/blob/4ec7077de5714cbafc18a5a807b3cdd020043b62/src/vanilla.ts#L61
